x-pack/plugins/security_solution/cypress/e2e/detection_rules/bulk_edit_rules_actions.cy.ts

```diff
@@ -0,0 +1,186 @@
+    esArchiverResetKibana();
+
+    createCustomRuleEnabled(getNewRule(), '1');
+    createEventCorrelationRule(getEqlRule(), '2');
+    createMachineLearningRule(getMachineLearningRule(), '3');
+    createCustomIndicatorRule(getNewThreatIndicatorRule(), '4');
+    createThresholdRule(getNewThresholdRule(), '5');
+    createNewTermsRule(getNewTermsRule(), '6');
+    createSavedQueryRule({ ...getNewRule(), savedId: 'mocked' }, '7');
+
+    createSlackConnector();
+  });
+
+  context('Restricted action privileges', () => {
+    it('User has no actions privileges', () => {
+      login(ROLES.hunter_no_actions);
+      visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL, ROLES.hunter_no_actions);
+      waitForRulesTableToBeLoaded();
+
+      selectNumberOfRules(expectedNumberOfCustomRulesToBeEdited);
+
+      openBulkActionsMenu();
+
+      cy.get(ADD_RULE_ACTIONS_MENU_ITEM).should('be.disabled');
+    });
+  });
+
+  context('All actions privileges', () => {
+    beforeEach(() => {
+      visitWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+      waitForRulesTableToBeLoaded();
+    });
+
+    it('Add rule actions to rules', () => {
+      const expectedActionFrequency = 'Daily';
+
+      loadPrebuiltDetectionRulesFromHeaderBtn();
+
+      // select both custom and prebuilt rules
+      selectNumberOfRules(expectedNumberOfRulesToBeEdited);
+      openBulkEditRuleActionsForm();
+
+      // ensure rule actions info callout displayed on the form
+      cy.get(RULES_BULK_EDIT_ACTIONS_INFO).should('be.visible');
+
+      pickActionFrequency(expectedActionFrequency);
+      addSlackRuleAction(expectedSlackMessage);
+
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfRulesToBeEdited });
+
+      // check if rule has been updated
+      goToEditRuleActionsSettings();
+
+      assertSelectedActionFrequency(expectedActionFrequency);
+      assertSlackRuleAction(expectedSlackMessage);
+    });
+
+    it('Overwrite rule actions in rules', () => {
+      const expectedActionFrequency = 'On each rule execution';
+
+      loadPrebuiltDetectionRulesFromHeaderBtn();
+
+      // select both custom and prebuilt rules
+      selectNumberOfRules(expectedNumberOfRulesToBeEdited);
+      openBulkEditRuleActionsForm();
+
+      pickActionFrequency(expectedActionFrequency);
+      addSlackRuleAction(expectedSlackMessage);
+
+      // check overwrite box, ensure warning is displayed
+      checkOverwriteRuleActionsCheckbox();
+      cy.get(RULES_BULK_EDIT_ACTIONS_WARNING).contains(
+        `You're about to overwrite rule actions for ${expectedNumberOfRulesToBeEdited} selected rules`
+      );
+
+      submitBulkEditForm();
+      waitForBulkEditActionToFinish({ rulesCount: expectedNumberOfRulesToBeEdited });
+
+      // check if rule has been updated
+      goToEditRuleActionsSettings();
+
+      assertSelectedActionFrequency(expectedActionFrequency);
+      assertSlackRuleAction(expectedSlackMessage);
```

Comment:
Let's also check that the rule after editing has exactly one action (no extra actions were added).